com.unity.shadergraph/Editor/Drawing/ViewModels/BlackboardViewModel.cs

```diff
@@ -2,22 +2,44 @@
 using System.Collections.Generic;
 using System.Linq;
 using UnityEngine;
+using UnityEngine.UIElements;
 using UnityEditor;
 
-namespace UnityEditor.ShaderGraph.Drawing.Views.Blackboard
+namespace UnityEditor.ShaderGraph.Drawing
 {
-    class BlackboardViewModel : SGViewModel
+    class BlackboardViewModel : ISGViewModel
     {
-        List<Type> m_ShaderInputTypes;
+        public GraphData model { get; set; }
+        public VisualElement parentView { get; set; }
+        public string title { get; set; }
+        public string subtitle { get; set; }
+        public Dictionary<string, IGraphDataAction> propertyNameToAddActionMap { get; set; }
+        public Dictionary<string, IGraphDataAction> defaultKeywordNameToAddActionMap { get; set; }
+        public Dictionary<string, IGraphDataAction> builtInKeywordNameToAddActionMap { get; set; }
 
-        public override void ConstructFromModel(GraphData graphData)
+        public Action<IGraphDataAction> requestModelChangeAction { get; set; }
+
+        public List<CategoryData> categoryInfoList { get; set; }
+
+        // Can't add disbled keywords, so don't need an add action
+        public List<string> disabledKeywordNameList { get; set; }
+
+        public BlackboardViewModel()
         {
-            m_ShaderInputTypes = TypeCache.GetTypesWithAttribute<BlackboardInputInfo>().ToList();
+            propertyNameToAddActionMap = new Dictionary<string, IGraphDataAction>();
+            defaultKeywordNameToAddActionMap = new Dictionary<string, IGraphDataAction>();
+            builtInKeywordNameToAddActionMap = new Dictionary<string, IGraphDataAction>();
+            categoryInfoList = new List<CategoryData>();
+            disabledKeywordNameList = new List<string>();
         }
 
-        public override void WriteToModel()
+        public void Reset()
         {
-            throw new System.NotImplementedException();
+            propertyNameToAddActionMap.Clear();
```

Comment:
Do you need to clear the other members like parentView and the titles here as well?

Reply:
Yes was missing some other members here, but regarding wiping the parentView, I have renamed this function to `ResetViewModelData` in the base interface to reflect its purpose better, which is that it should only be clearing out the data that the view directly uses in the presentation, as this function is called before a controller initializes the view model with the data it needs.

Also has to do with how the view models are currently fed the parentView reference only once, at their creation time, which might be something subject to debate, it made sense to me as we don't have dockable/undockable windows right now, but might in the future with overlays.